### ISSUE TITLE
Docker port re-mapping causes connection refused

### ISSUE THREAD

**Issue by thickconfusion** [Jackett/Jackett#9302]
### Environment

**OS**:  Docker version 19.03.12, build 48a6621 on Linux ubuntu 5.4.0-1013-raspi

**.Net Runtime**: [.Net-Core/.Net-Framework/Mono] All per linuxserver/jackett:latest 5 Aug 2020

**.Net Version**:  All per linuxserver/jackett:latest 5 Aug 2020

**Jackett Version**:  0.16.949.0

**Last Working Jackett Version**: N/A

**Are you using a proxy or VPN?** [yes/no] no

### Description

Using Docker and Docker-Compose, if I use all default settings for Jackett (port 9117 and External Access enabled), I can reach Jackett and API calls if I reach http://jackett:9117. (Docker -p 9117:9117). However, If I completely remove the container and settings and start over, this time mapping -p 9118:9117, I can reach the UI home page but can't make API calls.

1. -p 9117:9117, External access = enabled, Server port: 9117
```
root@6661a6964c5d:/# curl -vvv http://jackett:9117/api/v2.0/indexers/1337x/results/torznab/api?apikey=mbqtp601x6f37h60v81deicjdlwlyl1w&t=
search&cat=&q=
[1] 315
[2] 316
[3] 317
[2]-  Done                    t=search
root@6661a6964c5d:/# *   Trying 172.22.0.4...
* TCP_NODELAY set
* Connected to jackett (172.22.0.4) port 9117 (#0)
> GET /api/v2.0/indexers/1337x/results/torznab/api?apikey=mbqtp601x6f37h60v81deicjdlwlyl1w HTTP/1.1
> Host: jackett:9117
> User-Agent: curl/7.58.0
> Accept: */*
>
< HTTP/1.1 200 OK
```

2. -p 9118:9117, External access = enabled, Server port: 9117
```
root@ffd919596d09:/# curl -vvv http://jackett:9118/api/v2.0/indexers/1337x/results/torznab/api?apikey=mbqtp601x6f37h60v81deicjdlwlyl1w&t=
search&cat=&q=
[1] 299
[2] 300
[3] 301
root@ffd919596d09:/# *   Trying 172.23.0.2...
* TCP_NODELAY set
* connect to 172.23.0.2 port 9118 failed: Connection refused
* Failed to connect to jackett port 9118: Connection refused
* Closing connection 0
curl: (7) Failed to connect to jackett port 9118: Connection refused
```

3. (Same as above, then editing the configuration page to set server port to 9118):
```
root@ffd919596d09:/# curl -vvv http://jackett:9118/api/v2.0/indexers/1337x/results/torznab/api?apikey=mbqtp601x6f37h60v81deicjdlwlyl1w&t=
search&cat=&q=
[1] 309
[2] 310
[3] 311
[2]-  Done                    t=search
root@ffd919596d09:/# *   Trying 172.23.0.2...
* TCP_NODELAY set
* Connected to jackett (172.23.0.2) port 9118 (#0)
> GET /api/v2.0/indexers/1337x/results/torznab/api?apikey=mbqtp601x6f37h60v81deicjdlwlyl1w HTTP/1.1
> Host: jackett:9118
> User-Agent: curl/7.58.0
> Accept: */*
>
< HTTP/1.1 200 OK
```

**_But_**, I can no longer reach the server configuration page from the docker host at **http://localhost:9118/** CONNECTION REFUSED.

Here is the docker compose setup:

(network type is bridge)

```
  jackett:
    image: linuxserver/jackett:latest
    container_name: jackett
    restart: unless-stopped
    ports:
      - ${JACKETT_EXTPORT}:${JACKETT_INTPORT}
    volumes:
      - ${DOCKERDIR}/jackett:/config
      - ${DOCKERDIR}/Downloads:/downloads
    environment:
      PUID: $PUID
      PGID: $PGID
      TZ: $TZ
    security_opt:
      - no-new-privileges:true
```

### Logged Error Messages

See above.

### Screenshots
N/A
[Place any screenshots of the issue here if needed]

**Comment by ngosang** [comment]
@thickconfusion Jackett is working fine. I just tested it.
The problems is you are not understanding well how Docker works.
When you configure port mapping like this `-p 9118:9117`:
* Docker creates a new network interface and uses the container name as DNS. That means DNS jackett=172.23.0.2 (docker container IP)
* You can access your external/host IP+ Host Port => http://localhost:9118 or http://127.0.0.1:9118 or http://<server-lan-ip>:9118
* You can access your internal/docker IP + Docker Port => http://172.23.0.2:9117 (if you are inside the Docker containers you can user the container name => http://jackett:9117)

In you case you should use http://localhost:9118 (if you are in the same computer running Docker) or http://<server-lan-ip>:9118 (if you are running docker in a NAS/Server and you want to access from other computer)

**Comment by thickconfusion** [comment]
> - You can access your internal/docker IP + Docker Port => http://172.23.0.2:9117 (if you are inside the Docker containers you can user the container name => http://jackett:9117)

Thanks, this was the piece I was misunderstanding. I had been trying to use curl attempts from inside a sister container (Sonarr) to http://jackett:9118 since that's how I had mapped the port to host. These were giving connection refused. From that same sister container when I used http://jackett:9117, the connection succeeded. The point is, even though the docker host sees jackett's port 9117 traffic mapped to 9118, the other containers access Jackett's normal daemon port of 9117.
